src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/DeliverySessionController.kt

```diff
@@ -118,4 +133,88 @@ class DeliverySessionController(
     val user = userMapper.fromToken(authentication)
     return DeliverySessionDTO.from(deliverySessionService.submitSessionFeedback(actionPlanId, sessionNumber, user))
   }
+
+  @GetMapping("/referral/{referralId}/delivery-session-appointments/{appointmentId}")
+  fun getDeliverySessionAppointment(
+    @PathVariable(name = "referralId") referralId: UUID,
+    @PathVariable(name = "appointmentId") appointmentId: UUID,
+    authentication: JwtAuthenticationToken,
+  ): DeliverySessionAppointmentDTO {
+    val user = userMapper.fromToken(authentication)
+    val referral = referralService.getSentReferral(referralId) ?: throw ResponseStatusException(
+      HttpStatus.BAD_REQUEST, "sent referral not found [referralId=$referralId]"
+    )
+    referralAccessChecker.forUser(referral, user)
+    val matchingAppointment = deliverySessionService.getSessions(referralId)
+      .flatMap { session -> session.appointments.map { appointment -> Pair(session.sessionNumber, appointment) } }
+      .firstOrNull { appointment -> appointment.second.id == appointmentId }
+      ?: throw EntityNotFoundException("Delivery session appointment not found [referralId=$referralId, appointmentId=$appointmentId]")
+    return DeliverySessionAppointmentDTO.from(matchingAppointment.first, matchingAppointment.second)
+  }
+
+  @GetMapping("/referral/{referralId}/delivery-session-appointments")
+  fun getDeliverySessionAppointments(
+    @PathVariable(name = "referralId") referralId: UUID,
+    authentication: JwtAuthenticationToken,
+  ): List<DeliverySessionAppointmentDTO> {
+    val user = userMapper.fromToken(authentication)
+    val referral = referralService.getSentReferral(referralId) ?: throw ResponseStatusException(
+      HttpStatus.BAD_REQUEST, "sent referral not found [referralId=$referralId]"
+    )
+    referralAccessChecker.forUser(referral, user)
+    return deliverySessionService.getSessions(referralId)
+      .flatMap { session -> session.appointments.map { appointment -> DeliverySessionAppointmentDTO.from(session.sessionNumber, appointment) } }
+  }
+
+  @PostMapping("/referral/{referralId}/delivery-session-appointments")
+  fun scheduleNewDeliverySessionAppointment(
```

Comment:
Would the word "initial" be clearer as "new" appointments are event created when rescheduling

Reply:
This is either the first (delivery session) appointment or any additional appointment.
If additional then it checks that the existing appointment is before this new one and that the feedback was submitted.